packages/scrollable/src/react/index.tsx

```diff
@@ -0,0 +1,204 @@
+    useImperativeHandle(forwardedRef, () => ref.current as HTMLElement)
+
+    const dragPreview = useRef(BLANK_IMAGE)
+
+    const [hidden, setHidden] = useState<boolean>(true)
+    const [pageY, setPageY] = useState<number>(0)
+    const [offset, setOffset] = useState<string>('0%')
+    const [scrollRatio, setScrollRatio] = useState<number>(0)
+
+    const updateDimensions = useCallback(() => {
+      if (!ref.current) return
+      const content = ref.current
+
+      const nextTotalHeight = content.scrollHeight
+      const nextOffset = (content.scrollTop / nextTotalHeight) * 100 + '%'
+      const nextScrollRatio = content.clientHeight / nextTotalHeight
+      const shouldHide = nextScrollRatio >= 1
+
+      setHidden(shouldHide)
+      setOffset(nextOffset)
+      setScrollRatio(nextScrollRatio)
+    }, [ref])
+
+    useResizeObserver(ref, updateDimensions)
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    useEffect(() => updateDimensions(), [])
+
+    function onHandleDrag(evt: React.DragEvent<HTMLDivElement>) {
+      const nextPageY = evt.pageY
+      if (nextPageY <= 0) return
+
+      const content = ref.current
+      if (!content) return
+      const delta = evt.pageY - pageY
+
+      setPageY(nextPageY)
+      content.scrollTop += delta / scrollRatio
+    }
+
+    function onHandleDragEnd(_evt: React.DragEvent<HTMLDivElement>) {
+      setPageY(0)
+    }
+
+    function onHandleDragStart(evt: React.DragEvent<HTMLDivElement>) {
+      if (
+        evt &&
+        evt.dataTransfer &&
+        evt.dataTransfer.setDragImage &&
+        dragPreview.current
+      ) {
+        evt.dataTransfer.setDragImage(dragPreview.current, 0, 0)
+      }
+
+      setPageY(evt.pageY)
+    }
+
+    const styleOverride = useMemo(() => {
+      const height = Math.max(scrollRatio * 100, 10) + '%'
+      const visibility = hidden ? 'hidden' : 'initial'
+
+      return css({ height, top: offset, visibility })
+    }, [hidden, offset, scrollRatio])
+
+    return (
+      <Outer data-scrollable {...rest}>
+        <Inner key="scrollable-wrapper">
+          {renderContent({
+            children: props.children,
+            cssSelectors: styles.content(),
+            onScroll: updateDimensions,
+            setRef: (node: HTMLElement) => (ref.current = node)
+          })}
+        </Inner>
+
+        <Handle
+          draggable
+          onDrag={onHandleDrag}
+          onDragEnd={onHandleDragEnd}
+          onDragStart={onHandleDragStart}
+          {...styleOverride}
+        />
+      </Outer>
+    )
+  }
+)
+Scrollable.displayName = 'Scrollable'
+
+interface renderContentProps {
+  children: React.ReactNode
+  cssSelectors: StyleAttribute
+  onScroll: React.UIEventHandler
+  setRef: React.RefCallback<HTMLElement>
+}
+const defaultRenderContent = (props: renderContentProps) => {
+  return (
+    <div {...props.cssSelectors} onScroll={props.onScroll} ref={props.setRef}>
+      {props.children}
+    </div>
+  )
+}
```

Comment:
maybe there is a way to do the css different? i like the idea of just spreading all the props like

`(contentProps, contentRef) => <div ref={ref} {...contentProps} />`

i think that change would help usage and upgrades

Reply:
Yeah, I'll play with this.  Thanks!